### ISSUE TITLE
分かち書き発見要求仕様と発見機構仕様の両方: 作者指定の分かち書き候補位置があるかどうか

### ISSUE THREAD

**Issue by murata2makoto** [Japan-Daisy-Consortium/Technical-Committee#20]
分かち書き可能かというのは、実質的に意味がほとんどありません。自動処理によって分かち書き位置を決定し、分かち書きすることはほぼ常に可能だからです。

作者指定の分かち書き候補位置があるものを捜せることを、要求項目とすべきだと思います。実現機構仕様についても、それに基づいて直すべきです。

なお、作者指定の分かち書き候補位置だけでしか分かち書きしないという意味ではありません。自動処理によってさらに分かち書き候補位置を追加する（たとえば名詞と助詞の間）ことは可能ですし、実装がそういう工夫をすることはむしろ推奨されるべきでしょう。

**Comment by t-kudou** [comment]
分かち書き位置が著者や出版社、製作者が責任を持って入れたことをどのように表現すべきかが残課題です。
11/21の技術委員会では「正確性」という表現は適切ではないということになりました。

**Comment by kzakza** [comment]
分かち書きについては、以下の2つがあり得るという話でした。

・作者指定の分かち位置をwbrで指定する（RSがわかち書きの表示非表示を切り替えられる）
・スペース区切りでわかちを表現したもの

メタデータとしては、上のいずれも分かち書き位置を著者や出版社、製作者側が指定しているという点では同じと考えます。
これらをまとめて

・分かち書き表示が可能なEPUBに対して付与するメタデータ 
`<meta property="schema:accessibilityFeature">wordDividers/display</meta> `

で表すとすると考えて良いのではないかと思います。一方で以下のnon-displayは不要ではないかと思いました。

・分かち書きされていない表示が可能なEPUBに対して付与するメタデータ 
`<meta property="schema:accessibilityFeature">wordDivider/non-display</meta> `

理由ですが、上は以下の場合において、RSが分かち非表示ができることを表すとのことですが、RSがそのような機能を備えているということであり、コンテンツのメタデータとして表すものではないように感じました。
・作者指定の分かち位置をwbrで指定する（RSがわかち書きの表示非表示を切り替えられる）

強いていうなら、以下の2つをつけることで、分かちの表示非表示切り替えが可能はRSを前提としてwbrが埋め込まれているという状態を表すことができるのかもしれません。

`<meta property="schema:accessibilityFeature">wordDividers/display</meta> `
`<meta property="schema:accessibilityFeature">wordDivider/non-display</meta> 

しかし、それを表すならば、以下のそれぞれに値を与えたほうがシンプルではないかと思いました。

・作者指定の分かち位置をwbrで指定する（RSがわかち書きの表示非表示を切り替えられる）
・スペース区切りでわかちを表現したもの

ただ、メタデータは一度決まると、10年、20年は普通に使っていくものなので、あまり具体的な実装を前提とした値は望ましくないと思います。現在は、分かち書きを表す方法がスペース区切りしかないため、その手法が使われていると理解しています。
わかち書きの表示非表示を切り替えられるRSが当たり前になれば、wbrで指定するに移行するような気もしますし。

**Comment by t-kudou** [comment]
non-displayが無いと全角スペースで分かち書きが行われており、RSでは分かち書き部分を非表示にできないEPUBタイトルを表現できません。児童書のように分かち書きした状態を強く前面に出したい本は、wbrの表現が可能であったとしても、全角スペースを使った表現が使われるように思います。
また、詩や俳句のように全角スペースが重要な意味を持つようなものまでwbrタグで分かち書きされるとは考えにくいです。
このような意味で、non-displayは必要だと感じます。

**Comment by kzakza** [comment]
なるほどです。

non-displayは、「分かち書きされていない表示が可能」という意味だったかと思いますが、分かち書きを非表示にすること自体はRSの機能であって、コンテンツの機能ではないので、コンテンツに対するメタデータとして良いのかが若干ひっかかります。

工藤さんのご意図だと、全角スペースで分かち書きしていて、「分かち書きされていない表示が不可能」をメタデータとして表したいというというものだと思うのですが、「分かち書きされていない表示が不可能」なら、コンテンツに対するメタデータの情報としてすっきりするのですが・・。

**Comment by t-kudou** [comment]
「分かち書きされていない表示が不可能」をメタデータに埋め込む場合、利用者側からすると「分かち書きされていない表示が不可能」というくくりで検索することになり、とても判りにくくなります。
利用者側のニーズとしては単に、分かち書きされていない本を探したいということなので、利用者側のニーズに寄せたメタデータの方が良いと考えた次第です。

**Comment by murata2makoto** [comment]
今は、次の二点に絞って検討しています。

- A: 正確な分かち書き表示が可能か不可能か
- B: 分かち書きされていない表示が可能か不可能か

結局四つの場合に分かれます。

**場合1: Aが可能かつBが可能**

正確な分かち書き表示が可能
分かち書きされていない表示が可能

wbrが埋めこまれているタイプ。望ましい。

**場合2: Aが可能かつBが不可能**

正確な分かち書き表示が可能
分かち書きされていない表示が不可能

スペースが埋めこまれているタイプ。児童書など。

**場合3: Aが不可能かつBが可能**

正確な分かち書き表示が不可能
分かち書きされていない表示が可能

普通の出版物はこれになる。

**場合4: Aが不可能かつBが不可能**

正確な分かち書き表示が不可能
分かち書きされていない表示が不可能

これはありえない。

場合4は有り得ないし、典型的なのは場合3です。
となるとメタデータでは、

正確な分かち書き表示が指定されているが分かち書きしないことも可能なもの
正確な分かち書き表示が指定されていて分かち書きが必ず行われるもの

の二通りだけ指定できればいいように思います。

**Comment by t-kudou** [comment]
通常の出版物は場合3になりますが、こちらはメタデータを付与する必要はないでしょうか。

メタデータが3パターンできあがりますが、分かち書き表示をしたい本を探す場合は、場合1と場合2をor検索する必要があります。
縦組・横組の議論と被りますが、検索のシステムがor検索できるというのを前提条件にしなければならなくなり実装に負担がかかります。

**Comment by murata2makoto** [comment]
>通常の出版物は場合3になりますが、こちらはメタデータを付与する必要はないでしょうか。

ないと思います。

きちんとした指定をしてくれるのは、手間をかけて分かち書きを作りこむ人ぐらいでしょうから、それ以外の人にハードルを上げると普及が難しくなります。したがって、メタデータを明示的に指定しないときは場合3になるようにすべきです。

**Comment by t-kudou** [comment]
メタデータを付与しない場合には場合3である旨を明記するということで良いと思います。

分かち書きされた本を探すためには、OR検索が実装面に必ず要求される点についてはいかがでしょうか。

**Comment by murata2makoto** [comment]
>分かち書きされた本を探すためには、OR検索が実装面に必ず要求される点についてはいかがでしょうか。

出版社・取次・電子書店というサプライチェーンの運用の簡単さがもっとも重要であり、検索の実装の容易さはそれより優先度はずっと低いと認識しています。

**Comment by t-kudou** [comment]
了解しました。
次のようなメタデータの指定方法はどうでしょうか。

正確な分かち書き表示が指定されているが分かち書きしないことも可能なもの
wordDivider/nonmandatory

正確な分かち書き表示が指定されていて分かち書きが必ず行われるもの
wordDivider/mandatory
